access/errors.go

```diff
@@ -0,0 +1,189 @@
+package access
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/module/irrecoverable"
+)
+
+// RequireNoError returns nil if error is nil, otherwise throws an irrecoverable exception
+func RequireNoError(ctx context.Context, err error) error {
+	if err == nil {
+		return nil
+	}
+
+	irrecoverable.Throw(ctx, err)
+	return irrecoverable.NewException(err)
+}
+
+// RequireErrorIs returns the error if it unwraps to any of the provided target error types
+// Otherwise, it throws an irrecoverable exception
+func RequireErrorIs(ctx context.Context, err error, targetErrs ...error) error {
+	if err == nil {
+		return nil
+	}
+
+	for _, targetErr := range targetErrs {
+		if errors.Is(err, targetErr) {
+			return err
+		}
+	}
+
+	irrecoverable.Throw(ctx, err)
+	return irrecoverable.NewException(err)
+}
+
+// RequireAccessError returns the error if it is an Access sentinel error
+// Otherwise, it throws an irrecoverable exception
+// Note: this method will not unwrap the error. the passed error must be an instance of a sentinel
+// error.
+func RequireAccessError(ctx context.Context, err error) error {
```

Comment:
This can be used for more complex endpoints that call into other methods to ensure all unexpected errors are handled.

Reply:
you could add [your comment](https://github.com/onflow/flow-go/pull/7106#discussion_r1980178203) as context to the goDoc.